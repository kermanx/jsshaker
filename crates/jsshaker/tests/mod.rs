use std::fs;

use insta::{assert_snapshot, glob};
use jsshaker::{JsShakerOptions, TreeShakeConfig, tree_shake, vfs::SingleFileFs};
use oxc::{
  codegen::{CodegenOptions, CommentOptions},
  minifier::MinifierOptions,
};

fn do_tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = tree_shake(
    JsShakerOptions {
      vfs: SingleFileFs(input),
      config: TreeShakeConfig::recommended().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions { mangle: None, ..Default::default() }),
      codegen_options: CodegenOptions { comments: CommentOptions::default(), ..Default::default() },
    },
    SingleFileFs::ENTRY_PATH.to_string(),
  );
  result.codegen_return[SingleFileFs::ENTRY_PATH].code.clone()
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(do_tree_shake(input));
    })
  });
}
