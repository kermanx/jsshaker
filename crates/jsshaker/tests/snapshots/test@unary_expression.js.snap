---
source: crates/jsshaker/tests/mod.rs
expression: do_tree_shake(input)
input_file: crates/jsshaker/tests/fixtures/unary_expression.js
---
export const result = {
	s1: "number",
	s2: "bigint",
	s3: "string",
	s4: "boolean",
	s5: "object",
	s6: "undefined",
	s7: "object",
	s9: typeof f1,
	s11: typeof unknown,
	s12: (unknown, "string"),
	s13: typeof (unknown ? "_" : 1)
};
export function deletion() {
	const obj1 = { c: 3 };
	if (unknown) {
		delete obj1.c;
	}
	const c = unknown;
	delete unknown[c];
	effect(void 0, void 0, obj1.c, 4);
}
